src/server/.tests/ws-routes/controllers/Friends.spec.js

```diff
@@ -0,0 +1,497 @@
+						suite.emitSpy = sinon.spy(socket, 'emit')
+						return suite.friendsInstance.pendingNotifications(data, socket)
+							.then(() => {
+								//then
+								const expectedAttachment = QUERY_FEEDBACK_MOCK.pendingNotifications
+								sinon.assert.calledWith(suite.emitSpy.firstCall, CLIENT_EVENTS.PENDING_NOTIFICATIONS, expectedAttachment)
+								done()
+							})
+					})
+				})
+
+				//when
+				suite.client = socketClient.connect(SERVER_URL, SOCKET_OPTIONS)
+				suite.client.emit(CLIENT_EVENTS.PENDING_NOTIFICATIONS)
+			})
+		})
+
+		describe('#removeNotificationsFromList', () => {
+			it('should remove notification from db', () => {
+				//given
+				suite.userModelMock.findOneAndUpdate.returns({
+					exec: sinon.stub()
+				})
+				const requestingUsername = suite.DUMMY_USERNAME
+				const NOTIFICATION_ID_LIST =  [
+					{
+						_id: 'dummyId1'
+					},
+					{
+						_id: 'dummyId2'
+					}
+				]
+
+				//when
+				suite.friendsInstance.removeNotificationsfromList(NOTIFICATION_ID_LIST, requestingUsername)
+
+				//then
+				const expectedSearchingCirteria = {username: suite.DUMMY_USERNAME}
+				const expectedQuery = {
+					$pull: {
+						pendingNotifications: {
+							$or: NOTIFICATION_ID_LIST
+						}
+					}
+				}
+
+				sinon.assert.calledWith(suite.userModelMock.findOneAndUpdate.firstCall, expectedSearchingCirteria, expectedQuery)
+			})
+		})
+	})
+
+	describe('#Invite', () => {
+		beforeEach(() => {
+			suite.DUMMY_INVITATED_USERNAME = 'DUMMY_USERNAME_2'
+			suite.friendsInstance = new FriendsProvider({
+				UserModel: suite.userModelMock
+			})
+			suite.addNotificationStub = sinon.stub(suite.friendsInstance, 'addNotification').resolves()
+		})
+		it('should save notification with invitation in pendingNotificaiton array in invitated user document ', done => {
+			//given
+			const REQUEST_MOCK = {username: suite.DUMMY_INVITATED_USERNAME}
+			
+			suite.server.on(CLIENT_EVENTS.CONNECTION, socket => {
+				socket.on(CLIENT_EVENTS.INVITE, data => {
+					suite.friendsInstance.invite(data, socket, suite.connectionsMock)
+					then()
+				})
+			})
+
+			//when
+			suite.client = socketClient.connect(SERVER_URL, SOCKET_OPTIONS)
+			suite.client.emit(CLIENT_EVENTS.INVITE, REQUEST_MOCK)
+
+			//then
+			function then() {
+				const expectedInvitatingUsername = suite.DUMMY_USERNAME
+				const expectedInvitatedUsername = suite.DUMMY_INVITATED_USERNAME
+				const expectedEventType = CLIENT_EVENTS.INVITE
+				sinon.assert.calledWith(suite.addNotificationStub.firstCall, expectedInvitatingUsername, expectedInvitatedUsername, expectedEventType)
+				done()
+			}
+		})
+		
+		it('should send message to invited user with invite event type and attached payload with invitating username', done => {
+			//given
+			const REQUEST_MOCK = {username: suite.DUMMY_INVITATED_USERNAME}
+			suite.sendMessageToSpecyficUserSpy = sinon.spy(suite.friendsInstance, 'sendMessageToSepcificUser')
```

Comment:
Are there tests that check `sendMessageToSepcificUser` method?

Reply:
ok